table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java

```diff
@@ -0,0 +1,389 @@
+          mOwnerName,
+          mOwnerType,
+          glueDbDescription,
+          glueParameters);
+    } catch (EntityNotFoundException e) {
+      throw new IOException("Cannot find glue database" + mGlueDbName
+          + "." + e.getMessage(), e);
+    }
+  }
+
+  /**
+   * This method allows user to test udb glue client with remote glue server.
+   *
+   * @param config udbconfiguration
+   * @return glue async client
+   */
+  @VisibleForTesting
+  protected static AWSGlueAsync createAsyncGlueClient(UdbConfiguration config) {
+    ClientConfiguration clientConfig = new ClientConfiguration()
+        .withMaxConnections(config.getInt(Property.MAX_GLUE_CONNECTION));
+    AWSGlueAsyncClientBuilder asyncClientBuilder = AWSGlueAsyncClientBuilder
+        .standard()
+        .withClientConfiguration(clientConfig);
+
+    if (!config.get(Property.GLUE_REGION).isEmpty()) {
+      LOG.info("Set Glue region: {}.", config.get(Property.GLUE_REGION));
+      asyncClientBuilder.setRegion(config.get(Property.GLUE_REGION));
+    }
+
+    if (!config.get(Property.AWS_GLUE_ACCESS_KEY).isEmpty()) {
+      LOG.warn("Please setup the AWS access key id.");
+    }
+
+    if (!config.get(Property.AWS_GLUE_SECRET_KEY).isEmpty()) {
+      LOG.warn("Please setup the AWS access secret key.");
+    }
+
+    asyncClientBuilder.setCredentials(getAWSCredentialsProvider(config));
+
+    return asyncClientBuilder.build();
+  }
+
+  private static AWSCredentialsProvider getAWSCredentialsProvider(UdbConfiguration config) {
+    //TODO(shouwei): add compelete authentication method for glue udb
+    if (!config.get(Property.AWS_GLUE_ACCESS_KEY).isEmpty()
+        && !config.get(Property.AWS_GLUE_SECRET_KEY).isEmpty()) {
+      return new AWSStaticCredentialsProvider(
+        new BasicAWSCredentials(
+          config.get(Property.AWS_GLUE_ACCESS_KEY),
+          config.get(Property.AWS_GLUE_SECRET_KEY)));
+    }
+    return DefaultAWSCredentialsProviderChain.getInstance();
+  }
+
+  @Override
+  public String getType() {
+    return GlueDatabaseFactory.TYPE;
+  }
+
+  @Override
+  public String getName() {
+    return mGlueDbName;
+  }
+
+  @Override
+  public List<String> getTableNames() throws IOException {
+    try {
+      String nextToken = null;
+      List<String> tableNames = new ArrayList<>();
+      do {
+        GetTablesRequest tablesRequest =
+            new GetTablesRequest()
+                .withCatalogId(mGlueConfiguration.get(Property.CATALOG_ID))
+                .withDatabaseName(mGlueDbName)
+                .withNextToken(nextToken);
+        GetTablesResult tablesResult = mGlueClient.getTables(tablesRequest);
+        tablesResult.getTableList().forEach(table -> tableNames.add(table.getName()));
+        nextToken = tablesResult.getNextToken();
+      } while (nextToken != null);
+      return tableNames;
+    } catch (EntityNotFoundException e) {
+      throw new IOException("Failed to get glue tables: " + e.getMessage(), e);
```

Comment:
can you also add catalog id and database name into the message?

Reply:
The mountAlluxioPath is already moved to UdbUtils. I can add the catalog id and database name into the message.

Reply:
Can you add the db name and catalog id?